### DISCUSSION TITLE
Socked Keeps Disconnecting In Browser

### DISCUSSION THREAD

**Discussion by white5354** [eclipse-theia/theia#14134] [Q&A] [answered]
When i run [theia-blueprint](https://github.com/eclipse-theia/theia-blueprint) 
`yarn browser start`
**As Below**
![1](https://github.com/user-attachments/assets/5ed7a98c-39b5-46dc-a27e-8e53366c6208)

![微信截图_20240905165935](https://github.com/user-attachments/assets/89789cb9-d724-4000-8d1c-8e4c3e8522dc)

After starting the browser method, the socket keeps disconnecting and the page hangs without understanding.

What should I do?

**Comment by msujew** [comment]
Hey @white5354,

are there some more logs in the browser console of the frontend? The backend can tell relatively little if the connection fails.

**Reply by white5354** [reply]
@msujew thx for the reply
![image](https://github.com/user-attachments/assets/f4a4d6b6-b8c0-4897-ab19-c60b4373deea)

I guess I already know the problem. I created a new extension earlier.Maybe my extension's error.

**Reply by white5354** [reply]
@msujew I encountering new problems
Below this
![image](https://github.com/user-attachments/assets/90f0a2b1-32dc-42c2-8ffd-c2159a5c3fe4)

and my code 
```
import { injectable, inject } from '@theia/core/shared/inversify';
import { Command, CommandContribution, CommandRegistry, MenuContribution, MenuModelRegistry, MessageService, MAIN_MENU_BAR, nls  } from '@theia/core/lib/common';
import { CommonCommands, CommonMenus, KeybindingContribution, KeybindingRegistry } from '@theia/core/lib/browser';

export const MenuCustomCommand: Command = {
    id: 'MenuCustom.command',
    label: 'Say Good'
};

@injectable()
export class MenuCustomCommandContribution implements CommandContribution {

    constructor(
        @inject(MessageService) private readonly messageService: MessageService,
    ) { }

    registerCommands(registry: CommandRegistry): void {
        registry.registerCommand(MenuCustomCommand, {
            execute: () => this.messageService.info('Hello World! IDE')
        });
    }
}

@injectable()
export class MenuCustomContribution implements MenuContribution {

    registerMenus(menus: MenuModelRegistry): void {
        const newMenu = [...MAIN_MENU_BAR, '10_Say Other']
        menus.registerSubmenu(newMenu, nls.localizeByDefault('other'))
        menus.registerMenuAction(newMenu, {
            commandId: MenuCustomCommand.id,
            label: MenuCustomCommand.label
        });
        menus.unregisterMenuAction(CommonCommands.ABOUT_COMMAND)
        menus.unregisterMenuNode(CommonMenus.HELP[1])
        menus.registerSubmenu(CommonMenus.FILE, nls.localizeByDefault('文件'))
        
    }
}

@injectable()
export class MenuCustomKeybindingContribution implements KeybindingContribution {
    registerKeybindings(keybindings: KeybindingRegistry): void {
        keybindings.registerKeybinding({
            command: MenuCustomCommand.id,
            keybinding: 'ctrl+alt+h',
        });
    }
}
```
This code in [hello-world-Example](https://github.com/eclipse-theia/generator-theia-extension/blob/master/templates/hello-world/README.md) is Ok.
I don't know what the problem is.

**Reply by msujew** [reply, marked as answer]
@white5354 I believe you're pulling different versions of `@theia/core`. Can you run `yarn why @theia/core` and see whether it returns multiple results? Can you pin all versions of all `@theia/*` packages to **one exact** version?

**Reply by white5354** [reply]
@msujew You are right. I have two version. 1.53.1 and 1.53.0. I will try it your way

**Reply by white5354** [reply]
@msujew 😢  It didn't work. 
![image](https://github.com/user-attachments/assets/3b12f8c5-3d3e-42cf-b7ac-9e1b9fe5ec92)
Now it versions is 1.53.0. But problem still in existence.

**Reply by msujew** [reply]
Did you rebuild the frontend app? Does it still show the same inversify error?

**Reply by white5354** [reply]
wowww @msujew I forget rebuild.when i rebuild. It works. Thank you very much.I've learnt a lot.

**Reply by white5354** [reply]
I guess this issues can be closed @msujew
